src/main/java/me/axieum/mcmod/authme/api/util/MicrosoftUtils.java

```diff
@@ -64,14 +65,17 @@ public final class MicrosoftUtils
         .build();
 
     // Default URLs used in the configuration.
-    public static final String CLIENT_ID = "e16699bb-2aa8-46da-b5e3-45cbcce29091";
+    public static final String CLIENT_ID = "e16699bb-2aa8-46da-b5e3-45cbcce29091"; // AuthMe ClientID
     public static final String AUTHORIZE_URL = "https://login.microsoftonline.com/consumers/oauth2/v2.0/authorize";
     public static final String TOKEN_URL = "https://login.microsoftonline.com/consumers/oauth2/v2.0/token";
     public static final String XBOX_AUTH_URL = "https://user.auth.xboxlive.com/user/authenticate";
     public static final String XBOX_XSTS_URL = "https://xsts.auth.xboxlive.com/xsts/authorize";
     public static final String MC_AUTH_URL = "https://api.minecraftservices.com/authentication/login_with_xbox";
     public static final String MC_PROFILE_URL = "https://api.minecraftservices.com/minecraft/profile";
 
+    private static String refreshToken = null;
```

Comment:
This refresh token will only persist for the current Minecraft session - if that fixes logging into the same account again, I'm happy to merge this. But we will need to check the refresh token's expiry.

The Minecraft session shouldn't invalidate often, unless you leave the game open for a long time. So requesting a full token from scratch every time seemed reasonable at the time.

Aside: _I don't want to save access or refresh tokens to the config, unless they are saved to a file that isn't in `config/`. I just don't feel comfortable being responsible for people accidentally leaking their account.

_I've already seen some big modpacks with this mod's config distributed._

Reply:
> The default lifetime for the refresh tokens is 24 hours for single page apps and 90 days for all other scenarios.
Even if the token expires (after 90 days), it's probably possible to fall back to normal authorization if refresh tokens fail.

And I'm not really worried abt different Minecraft sessions, since the user logs in again after opening MC, creating a completely new session.

Reply:
I also don't really wanna mess around with storing the tokens between sessions as well, exactly for the same reason of having people accidentally leak the token.